JASP-Engine/JASP/R/correlation.R

```diff
@@ -1079,57 +981,69 @@ Correlation <- function(dataset=NULL, options, perform="run", callback=function(
 }
 
 ### Utility functions for nonparametric confidence intervals ###
-.ConcordanceFunction <- function(i, j)
-{
-  Q.ij <- 0
+.ConcordanceFunction <- function(i, j) {
+  concordanceIndicator <- 0
   ij <- (j[2] - i[2]) * (j[1] - i[1])
-  if(ij > 0) Q.ij <- 1
-  if(ij < 0) Q.ij <- -1
-  return(Q.ij)
+  if (ij > 0) concordanceIndicator <- 1
+  if (ij < 0) concordanceIndicator <- -1
+  return(concordanceIndicator)
 }
 
-.addConcordances <- function(x, y, i)
-{
-  C.i <- 0
-  for(k in 1:length(x))
-    if(k != i) 
-      C.i <- C.i + .ConcordanceFunction(c(x[i], y[i]), c(x[k], y[k]))
-    return(C.i)
+.addConcordances <- function(x, y, i) {
+  concordanceIndex <- 0
+  for (k in 1:length(x)) {
+    if (k != i) {
+      concordanceIndex <- concordanceIndex + .ConcordanceFunction(c(x[i], y[i]), c(x[k], y[k]))
+    }
+  }
+  return(concordanceIndex)
 }
 
-.createNonparametricConfidenceIntervals <- function(x, y, hypothesis = "two-sided", confLevel = 0.95, method = "kendall"){
+.createNonparametricConfidenceIntervals <- function(x, y, obsCor, hypothesis = "two-sided", confLevel = 0.95, method = "kendall"){
+  # Based on sections 8.3 and 8.4 of Hollander, Wolfe & Chicken, Nonparametric Statistical Methods, 3e.
   alpha <- 1 - confLevel
+  missingIndices <- as.logical(is.na(x) + is.na(y)) # index those values that are missing
+  x <- x[!missingIndices] # remove those values
+  y <- y[!missingIndices]
   n <- length(x)
-  corValue <- cor.test(x, y, method=method)$estimate
-  if(method == "kendall" && sum(is.infinite(c(x,y)))==0){
-    c.i <- numeric(0)
-    for(i in 1:n){
-      c.i <- c(c.i, .addConcordances(x, y, i))
+  
+ if (method == "kendall") {
+    concordanceSums <- numeric(n)
+    for (i in 1:n) {
+      concordanceSums[i] <- .addConcordances(x, y, i)
     }
-    sigmaHatSq <- 2 * (n - 2) * var(c.i) / n / (n-1)
-    sigmaHatSq <- sigmaHatSq + 1 - (corValue)^2
+    sigmaHatSq <- 2 * (n - 2) * var(concordanceSums) / n / (n-1)
```

Comment:
The following is the same: 

sigmaHatSq <- 2 * (n - 2) * var(concordanceSums) / (n*(n-1) )

but this looks much more like 8.38 of p. 415